ui/dashboards/src/components/Panel/PanelHeader.tsx

```diff
@@ -116,3 +128,12 @@ const HeaderIconButton = styled(IconButton)(({ theme }) => ({
   borderRadius: theme.shape.borderRadius,
   padding: '4px',
 }));
+
+const HeaderActionWrapper = styled(Stack)(() => ({
+  // Adding back the negative margins from MUI's defaults for actions, so we
+  // avoid increasing the header size when actions are present while also being
+  // able to vertically center the actions.
+  // https://github.com/mui/material-ui/blob/master/packages/mui-material/src/CardHeader/CardHeader.js#L56-L58
+  marginTop: -4,
+  marginBottom: -4,
```

Comment:
See the "Why the adjustments to the margins exists" in the PR description to show why we need this. I avoided putting a gif in the inline comments because it makes it hard to focus on the code. :)

Reply:
Thanks for all the comments / info in description about why negative margin is needed! I definitely was confused until I saw the gif 😄